### DISCUSSION TITLE
Support for objects other than ObjectID/String as ids

### DISCUSSION THREAD

**Discussion by sean-kates** [meteor/meteor#12337] [Feature Requests]
Hello,

Couldn't find any other discussion on this one in feature requests, so apologizes if it's a duplicate.

I'm working on scaling up our application to handle much larger collection sizes, billions of records in some. We are currently using string UUIDs in the _id field. I was researching was to improve compression and performance, and one thing I saw was about using binary UUIDs instead of strings.

https://stackoverflow.com/questions/46327072/performance-gains-to-using-uuid-instead-of-its-string-representation-as-id-on-m

I toyed around with it, but it doesn't look like Meteor supports them.

```
Error: Meteor does not currently support objects other than ObjectID as ids
    at _IdMap.MongoID.idStringify [as _idStringify] (packages/mongo-id/id.js:78:11)
    at _IdMap.has (packages/id-map/id-map.js:30:22)
    at packages/mongo/mongo_driver.js:1104:30
    at /app/bundle/programs/server/npm/node_modules/meteor/promise/node_modules/meteor-promise/fiber_pool.js:43:40
```

Are there any plans to allow this in the future? It seems like supporting them would be beneficial for larger scale Meteor applications.

Thanks!

**Comment by Grubba27** [comment]
Hello @sean-kates! Good to see feature requests! have you seen  [this](https://docs.meteor.com/api/collections.html#Mongo-ObjectID), right? I would ask for the MongoDB Driver docs for this API that you are trying to use. I guess we can think a way to make the driver more exposable/flexible to the enduser

**Comment by radekmie** [comment]
I agree that Meteor should handle other forms of IDs. But first of all, we should start with making sure the `ObjectId` works as expected (I couldn't find the thread of @baryshok about it; [here's the repo](https://github.com/baryshok/meteor-test-add-type-for-object-id), [feature request A](https://github.com/meteor/meteor-feature-requests/issues/428), [feature request B](https://github.com/meteor/meteor-feature-requests/issues/371)). Then we would continue with MongoDB UUIDs ([discussion](https://github.com/meteor/meteor/discussions/11553)). Only then we would extend it to all IDs.

**Comment by sean-kates** [comment]
@radekmie Ah I didn't see this [discussion](https://github.com/meteor/meteor/discussions/11553), exactly what I was looking for.

I'll test with ObjectIds this week and see how the results look, definitely a good option for compression. The problem there is that we use UUID V5 as a uniqueness check, and I don't believe ObjectIds will accomplish the same thing.

We can probably close this one out, as the Mongo team seems to have the discussion covered over there.

**Comment by jamauro** [comment]
ObjectIds are kind of a pain imo. I would like to see support for `uuidv7` so we can sort by `_id`. Honestly it would probably make for a better default than `Random.id`. I think it would "just work" for minimongo and mongo. Is there any appetite for a PR to Meteor core?

**Reply by StorytellerCZ** [reply]
`uuidv7` looks interesting. I'm curious if you can choose that somehow in MongoDB or something like that to be the default. If yes, then I would be very much for migrating to it. If not then I still like the idea, but I would like to know from MongoDB guys if they can take advantage of the benefits.

**Reply by jamauro** [reply]
Well I was thinking we'd just store them as strings. Then I'd expect them to play nicely with all the meteor things. 

I suppose we could go the extra mile and store as binary in mongo and apply a conversion in meteor so developers could work with the string representation and minimongo would still work. 

Either way I believe it'd still retain the sorting benefit but we'd have to test it out.
